### ISSUE TITLE
Clean LaFleur data

### ISSUE THREAD

**Issue by huntallcott** [huntallcott/Lotteries#22]
The goal of this issue is to make sure that the LaFleur data are fully clean.
I have some questions based on @jackblun's great work in #20. Jack, sorry if I missed this somewhere else, and I know you've been emailing about some of these issues with Byron.

1. Missing data for specific states and years 
I ran this code (and I know you've done something similar already):
`use "$dropbox/Externals/Intermediate/incomeeffects_clean.dta", clear`
`gen NYears = 1/4`
`collapse (max) LastYear=year (min) FirstYear=year (sum) NYears, by(state)`

Montana is missing 2015 and one other year, DC is missing 2016. Maine is missing one quarter. Rhode Island missing after 2010 in incomeeffects_clean, but it is not missing in the original lafleurs data. WVA sales are very close to zero but not zero in 2015. 
Can you explain more? 

2. Possible missing data for specific games within states. Obviously games come and go, but I think we want to make sure that these game entry and exits are real. Two examples (although there are many):

* Does Idaho really not have pb in 2010 at all, or is this LaFleur’s missing data?
* Rhode Island (which is an outlier in our Great Recession event study) has several different VLT games over the sample. Is this real variation or just the LaFleur’s data? 

3. I continue to strongly recommend switching to standard state codes. It just makes everything easier, including making graphs with the existing data, etc.

4. I know we have time effects in the regs, but it’s generally prudent to deflate data over such a long time series. See GitHub/Lotteries/Code/Build for a short script to get the CPI data that (I just added it yesterday), including a reference to where you might download the data.

5. Other than this, @jackblun are there other issues with the LaFleur data that are worth checking on? 

Many thanks.

**Comment by jackblun** [comment]
Thanks for highlighting this. Agreed that cleaning needs more work. I've had a few back and forths with Byron on this as more issues emerge.

1. There are two problems here. One is that some state-years just are not present in the La Fleur's data. As you've seen, I've asked Byron whether there is any way to rectify this. The other is that I have dropped some of the states you mention for data reasons. Here's the section where I comment on this:
`drop if state == "NE" & year == 2008 // weirdly high, instant sales misreported?
drop if state == "OR" & year >= 2012 // no VLT data
drop if state == "RI"  & year >= 2011 // no VLT data
drop if state == "WV" & year >= 2010 // no VLT data
drop if state == "SD" & year == 2016 // no VLT data`
VLT is the main issue. For some states, VLT is a huge share, and then in some years its just dropped from the data entirely for later years. Here I am dropping state-quarters where there is no VLT data. I'll comment in this in my next response below.

2. I will ask Byron to give a sense of how complete he thinks the data is in terms of games. There are certainly some examples in the data which seem implausible, as you point out. Related to the above, I will also ask what their approach is re VLT inclusion, as that really matters for totals. I'll do this once he replies to my previous email.

3. I've now changed this.

4. Sure, will do when data cleaner.

5. I think we will inevitably be left with some measurement error in this data, but I imagine the above will help. Now that Byron isn't trying to sell us the data, I would hope he will be open about how confident he is on game inclusion. One the issues above are sorted, I will plot total sales by states and inspect manually, to see if anything else looks amiss.

**Comment by jackblun** [comment]
I've fixed a couple of small errors in the 2010 data for PA and ME. One puzzle is that RI and WVA have super high per capita sales. I'll see what Byron says re VLT, as I think this might be something to do with it.

**Comment by jackblun** [comment]
Update on the cleaning side:
1. I have found that much of the drop in 2010 can in fact be explained by poor data. In many states, there are games missing for this year. Some of these are not 'niche' games, but big games such as powerball.
2. More generally, I am increasingly sceptical that we are always seeing the 'full' set of games on offer in each state here, so the total sales figure will likely be misleading. In the analysis I have moved to game level analysis, which already is working much better.

**Comment by jackblun** [comment]
I think it will vary. Given previous experience with them, if a state-year is entirely missing, they may have just made a mistake copying the entire sheet over so we may get some of those back. For individual missing games within state-years, I would guess that it is less likely there will be a fix. The fact that I’ve found many missing games in 2010 suggests to me that someone was sloppy on data entry that year. I will however keep following up.

**Comment by jackblun** [comment]
@anjoung , I understand you may have some time and be able to help with the cleaning of the La Fleur's data. This would be really helpful. We have purchased data on weekly sales of each lottery game, at the state level. I have spent some time cleaning the data and have discovered several anomalies. These are so far:
- States which are missing for certain years
- States which have incorrectly calculated total sales figures (due to summing over the wrong columns)
- States which are missing some games for some years

The ideal data would be for years 1998-2016, all games in all states. As best we can, we want to document any deviations from this and correct them where possible. It would be really useful to have a document compiling these and any other errors you find. We can then contact the data providers to see if they have corrections, and can correct some (such as incorrect sums in the excel files) ourselves. Would you be able to create such a document for us?

To identify errors, a first thing to do is just to plot data over time and looking for discontinuities. I have identified most of the above by plotting total lottery sales over time for each state.

The deliverable is either a word doc or a simple spreadsheet, saying for each state whether there are any evident issues with the data. If there is an issue that we fix, it should be noted how we have made that fix. 

The code is all in the income_effects subfolder in the build folder. You can ignore the file cleaning the bea data, and look primarily at dataprep_lafleurs, and then also dataprep_incomeeffects. Within these cleaning files I've made notes indicating the cases of bad data I've already found, which should enter the document. I've also corrected a small number of the raw data files, duplicating them and adding '_corrected', which should be fairly clear in the code.

Is that clear? Happy to provide a lot more detail once you've had a chance to look at the existing cleaning code.

**Comment by anjoung** [comment]
@jackblun This all sounds good to me! Let me know if there are any additional items you need me to tackle

**Comment by jackblun** [comment]
@anjoung brilliant - thanks. Do get in touch if anything is unclear in the code.

**Comment by anjoung** [comment]
I've made a little bit of headway on this. I made the following [document](https://github.com/huntallcott/Lotteries/files/2789352/Data.inconsistencies.docx). The document compiles all the notes in the do-files. I also added missing data that I noticed. So far, I've only assessed whether a state is missing the variable _all_games_sales_. I collapse this variable by state and century-week. Then, I tabulate all the state-century-weeks for which there is missing data. 

The document, and any notes that I am taking, are being added to the "Notes" subfolder in the Dropbox. Also, after reading the wiki page, I made a "scratch" folder, so the code I am using to inspect the data is in the  folder "~/Code/Temp-Andrew Checking Data". The code is called "CHECKlafleuers.dta".

Also, I just have a few questions for @jackblun:
1. Should I be looking for missing data at the weekly or quarterly-level? 
2. Is the variable _all_games_sales_ simply the sum of the _sales_ variable over all games in a state-century-month?
3. Is there a way to identify if a game is a PC game? Additionally, does "PC" mean that a lottery ticket was bought online?
4. Does VTL stand for Video Terminal Lottery?

**Comment by jackblun** [comment]
Thanks for the quick work Andrew, the doc looks good. Could you note in this file whether states have lotteries at all? We don’t care if we are missing a state's lottery sales due to there not being any lotteries in that state over that year, but we do care if there are lotteries but they don’t appear in our data.

I’m not sure I understand what you mean by century in this setting though, could you elaborate?

On your questions:
1. You should look on the weekly level. From looking at the data, it seems that most of the states are either missing entire years, or games for the whole year, however we should note if states are indeed missing data for certain weeks in the year.
2. The variable all-games-sales is the sum of all sales for each week in each state, as drawn from the lafleurs raw data. This is not always correctly calculated, as sometimes in the raw data they calculate the sum over the wrong columns. This is what is fixed in the _corrected files.
3. The _pc variable is per capita. However since we don’t know where they are getting the population, I’m not using this. Instead I think we should work with the aggregate sales from La fleurs and do any per capita calculations ourselves, using population data drawn from other sources.
4. Correct.

**Comment by anjoung** [comment]
Got it! Thanks for the answers. When I am flagging states without any lotteries, do you mean just state lotteries, or state lotteries+Mega Mill+Powerball? 

By century, I just mean year.

**Comment by jackblun** [comment]
I mean states without any lotteries altogether. Some states have only
introduced lotteries recently, and some (eg Nevada) don't have them at all.
I believe the lotteries in the US page on Wikipedia has a list.

**Comment by jackblun** [comment]
@anjoung on the data cleaning front, would it be possible for you to replicate what I have done for 1997 onwards for years 1994, 1995 and 1996? We have the La Fleurs data for these years but haven't cleaned it.

This should be fairly easy and you can copy paste most of the code I have already used, but there is one initial hurdle. When I first tried importing these earliest years, there was something wrong with the files. The data is there, but there is a problem reading it into Stata. I had a little look and think the easiest way around this would be to convert all the sheets into csv then import in that format rather than using ```import excel```.

Does that make sense? Thanks again!

**Comment by jackblun** [comment]
Great. The priority should be one the first task, where you finish cleaning and writing up errors for post 1998.

**Comment by jackblun** [comment]
Sorry - mean post 1997.

**Comment by jackblun** [comment]
Hey @anjoung , do you have an update on this? Would be great to hear how you're getting on.

**Comment by bblockwood** [comment]
Hi @jackblun — a bit of info from my check-in meeting with @anjoung Friday: he is pushing forward on the memo, and is leaving a “current status” note at the top of the *Data inconsistencies.docx” memo to indicate where things stand. Andrew is currently splitting time between this project and tasks for his co-supervisor, though he’ll be able to move to this essentially full-time at the end of February.

**Comment by anjoung** [comment]
@jackblun Updated the status of the document to reflect that I found some errors in my coding that I need to resolve. I am planning on finishing this either tomorrow afternoon, or Wednesday afternoon. Please let me know if you need this done faster!

**Comment by anjoung** [comment]
@jackblun Just finished my first run through of the memo! I've attached a bunch of files to this post, but all of them are available in the Notes subfolder on Dropbox. The only exception is the do-file for this which is located in "~/Code/Temp-Andrew Checking Data/CHECKlafleuers.do". The log files are reported in the memo. Please let me know if _anything_ is unclear, or if you need further analysis!

**Memo:** [Data inconsistencies.docx](https://github.com/huntallcott/Lotteries/files/2809848/Data.inconsistencies.docx)
**Missing Data at State-Level: **[Missing State Level.txt](https://github.com/huntallcott/Lotteries/files/2809849/Missing.State.Level.txt)
**Possible Miscalculated Data: **[Possible errors.txt](https://github.com/huntallcott/Lotteries/files/2809847/Possible.errors.txt)
**Missing Game Data: **[Missing State-Game Level.txt](https://github.com/huntallcott/Lotteries/files/2809850/Missing.State-Game.Level.txt)

**Comment by jackblun** [comment]
Hey @anjoung , thanks for all this work. My main takeaway from this is that games do seem to drop in and out of the sample suspiciously quickly, so we are right to be doing a game-level analysis.

Two questions / comments:

1. I'm not sure I understand what you're doing when you calculate errors in calculating the main sales variable. For example, in California year 2000 week 52, Total Online + Instant = Total Sales, as should be the case. However you've listed it as potentially having an error. Could you clarify?

2. Given what you've found so far, it would be very useful to move onto potential errors in sales data within games within states. This will be difficult to spot, but an obvious thing to check is whether there are any huge data entry errors. You could do this just be inspecting distributions of sales and inspecting outliers. If you find any, these should be flagged in the cleaning file which feeds into the analysis. Is that clear?

Thanks again!

**Comment by huntallcott** [comment]
Thanks @anjoung.
A. Jack, it seems useful to continue keeping track of whether we should be asking LaFleur to look for these missing data, or whether we should think of these data as totally gone. 
B. Just a note -- when we go back to analysis at the game-by-quarter level for income effects, it may be good to think about using data that is "average sales per week for non-missing weeks," given that there seem to be missing data for some specific weeks within a quarter.
Thanks again.

**Comment by anjoung** [comment]
@jackblun To identify errors for the main sales variable, I started with the cleaned data. Then, I summed the "sales" var by state-week. Then, I take this summed variable and the main sales variable, find the difference, and round to the second decimal place. Is there some sales information that is not present in the cleaned data, but is available in the raw data? I'll take a second look at all my code, as well.

I'll start trying to identify errors at the game level.

**Comment by jackblun** [comment]
Hi @anjoung thanks for the update. With these kind of tasks it is usually really useful (if possible) to just eyeball the raw data to sense check your findings. Cheers

**Comment by anjoung** [comment]
@jackblun Thank you!! Happy to do it.

**Comment by anjoung** [comment]
Planning on reviewing each spreadsheet by eye to make sure things are being calculated correctly at least on that level. After having reviewed some of the spreadsheets, I am not 100% sure why my method is not working. The main sales var seems to be the sum of all columns for a given row, excluding per capita measurements and sometimes VLT numbers. Otherwise, any errors in calculation would be apparent by identifying missing data (which would get treated like a zero in the sums). I think that this is what I am doing? Only saying this because in case there is something else going on.

Also, I noticed in the cleaned data that we have duplicates at the state-game-date level. I literally just load lafleurs_clean and type the command _duplicates report game state year week_. The command reports that we have 2160 duplicates. These duplicates seem to have different main sales vars, so I am not sure where these are coming from. These duplicates explain a portion of the calculation errors that I reported. Are these duplicates expected?

**Comment by anjoung** [comment]
Did some digging and think that I've nailed most of the problems. There are 19 observations where I cannot replicate the main sales variable successfully. I haven't given up; just haven't gotten to observe these individual cases. I eliminated most of the problems by removing data that was incorrectly read-in as raw data. Also, there were a few cases where data was being incorrectly calculated. I'll fully document the errors that I found in the memo tomorrow. 

However, the duplicates remain and I figured out what is happening. Every single duplicate is at the last week of the year. What happened is that sometimes there is drawing on December 24th and a drawing on December 31st. Stata regards both dates as belonging to the 52nd week, so they end up coming up as duplicates. Not sure what to do in this case.
<img width="591" alt="capture" src="https://user-images.githubusercontent.com/30239050/52028455-15957700-24dd-11e9-855f-4cfc4059a5f3.PNG">

**Comment by jackblun** [comment]
@anjoung thanks, thats a great find re the weeks at the end of the year. I will think how best to deal with that, but it shouldn't be too difficult.

**Comment by anjoung** [comment]
Figured out the remaining issues. There was negative sales values that were not getting read correctly. Are there supposed to be negative values?

**Comment by jackblun** [comment]
There should not be negative values. How frequent are these? If not super frequent, we could drop state-game-years where they appear.

**Comment by anjoung** [comment]
They are not super frequent and usually super small. I'll adjust the cleaning do-file to take this into account, and add these cases to the memo. 

I've nailed down the cases where we cannot replicate the main sales variable using the game-specific sales info. I have four cases where this is not due to to duplicate dates. In one case, this is because I am missing the sales variable for one state-date (but have game sales info). In the other three, it is because of negative sales data. Remaining are 136 pairs of duplicate dates. 

Additionally, I noticed that there are some VLTs that are reporting gross sales, and some that are reporting "net" sales or "income" (which I assume is net sales). Should we be treating these differently?

**Comment by huntallcott** [comment]
@jackblun can you just put the negative values on the list of things to ask Mr. LaFleur? I would like to know whether to make these zeros vs. consider them as valid numbers. I imagine they could be valid, based on what we learned about occasional scratchoff returns, right?

**Comment by jackblun** [comment]
Hmm okay, thanks for noticing with the VLTs. I expect we may have to omit VLTs altogether. For now, can you flag which type is reported?

**Comment by jackblun** [comment]
@huntallcott thats true, however if these aren't reflecting actual weekly sales, we would probably want to drop them right? I'll put all these issues in an email to Mr LaFleur once Andrew is done.

**Comment by anjoung** [comment]
@jackblun The memo should be fully updated with all the new stuff. I've flagged the specific state-weeks for which we have issues regarding replicating the main sales var, as well as whether the issue is due to "duplicate dates" or a negative number. Also, added any problems that I found in the raw data that I corrected. I hand-corrected the raw data for two spreadsheets (MO 2003, and PA 2003). I preserved the original raw data and renamed the files so it is clear that I edited them. 

Let me know if there are any remaining items that I still need to address, or any points that still remain unclear. Otherwise, I am going to start digging into game-specific errors, and see if I can figure anything out.

**Comment by anjoung** [comment]
@jackblun I have not been able to make any progress on the game-specific errors. The memo currently records any errors on La Fleur's end 1998 onward. I removed any miscalculation errors that we were able to correct from the memo, but include any manual corrections for the sake of maintaining a record of those changes.

I think the three main things on my plate are (1) finding game-specific errors, (2) finding errors from 1994 to 1997, and (3) cleaning the jackpot data and merging it with the sales data. How would you like me to prioritize these tasks? I can shift gears into whatever is most useful for you.

**Comment by anjoung** [comment]
Sorry I want to clarify on game-specific errors. I have a comprehensive list in the memo of all the games where we are missing data in the way that I detailed in the memo. However, I have not assessed if we are missing data at the tails (e.g., when was each game officially started/ended and does this map onto the data we have available). Outside of the tails, I think that I've identified and listed every other case of missing data. Identifying issues at the tails will take a long time.

Additionally, I have not been able to go game-by-game and verify that we are indeed missing data, and that the issue is not that we have incorrectly read-in the data. I don't perceive this to be a problem, but just want to completely transparent about what remains on my plate on this front.

**Comment by jackblun** [comment]
@anjoung thanks for the update and all the work. I think it would be a good time to move onto (3), cleaning the jackpot data and merging it with the sales data. Tasks (1) and (2) are less of a priority. 

Regarding the cleaning of the jackpot data, I think the place to start would be with the large multi-state games. A great intermediate target would just be to have a dataset of megamillions and powerball jackpots, along with sales of these in each state where they operate. Does that sound reasonable? There are two immediate issues:
1) The weeks will not line up. This can be easily solved by approximating. Just see what dates are most common and line them up as best you can.
2) The draws are sometimes twice a week, but the sales data are weekly. This is a harder problem. For now, it would be great to keep everything in at the most detailed frequency possible rather than averaging out draws.

**Comment by anjoung** [comment]
Got it! I'll get to work on this ASAP

**Comment by anjoung** [comment]
Made some headway on making the PB and MM dataset. Just want to outline some issues that cropped up. 

First, for winning jackpots, Powerball records the amount claimed rather than the advertised jackpot. Moreover, for those who claim cash prizes, the data reports the cash amount claimed rather than the annuity amount. One possible solution is to turn to [Wisconsin](https://www.wilottery.com/lottogames/powerballAllhistory.aspx) which seems to only record the advertised jackpots. As an added side benefit, this data also runs back to 1992 (whereas PB ends in 1997). However, when I compared the jackpot amounts reported by Wisconsin and PB for non-winning amounts, there were consistent discrepancies. I have not dug into why these discrepancies exist, but will do so soon.

Second, as per @bblockwood's request, below I made a binned scatterplot of weekly sales (in millions) of Powerball tickets on the y-axis, and the jackpot amount (in millions) for each draw on the x-axis. Since draws are biweekly, I dealt with this by just assigning each draw a year-week and the sales data associated with that year-week. This means that I duplicated the sales data. For example, if I had a draw 04feb2019 and 07feb2019, then my sales figure for 2019w5 (1st week of Feb) would be assigned to both draw dates. If this was a silly approach, let me know! 

Finally, I just want to flag two smaller things. I had to manually calculate the probabilities of winning each prize tier. I am 95% sure that I did it correctly, since I was able to check my math for a few cases with an [outside source](https://en.wikipedia.org/wiki/Powerball). Want to just make sure I flagged this case of manual entry. Secondly, how should I deal with the PowerPlay games in our data? Should I drop them when considering Powerball, or should I add it into the sales data that I am using to make the figure below?

![graph](https://user-images.githubusercontent.com/30239050/52304385-a5657600-2960-11e9-9edf-30a561860895.png)

**Comment by jackblun** [comment]
As an update - I've emailed La Fleurs again based on the info @anjoung recorded in his memo. I'm focusing on entirely missing years and advice on how they approach VLT sales, will update when they provide more info.

**Comment by jackblun** [comment]
Hey @anjoung , another data cleaning issue has come up with the La Fleurs data I am hoping you'll be able to help with. 

We are trying to analyze the effect of income on lottery sales by using game-state level information. I've realized that in the La Fleurs data, there is some inconsistency in the names of games over years. For example, in many years there is a game in DC called "HotFive", and in other years there is one called "Hot5". Right now, these are being treated as different games, which is clearly wrong. 

The way to fix this will be to first do some basic things, such as:
- change all capitals to lower case
- replace all 'letterized' numbers with actual numbers (eg 5 instead of 'five')
- change common abbreviations to be consistent (eg limited to ltd)

You should be able to fix quite a few of the games like this, using subinstr in Stata. After that, there may be some fiddly individual fixes that will have to be done by hand. Unfortunately this is essential for our game-state level analysis to work, and is therefore fairly high priority.

Is this something you can help with? If so, could you edit dataprep_incomeeffects.do to fix this? Its in Lotteries⁩ ▸ ⁨Code⁩ ▸ ⁨Build⁩ ▸ ⁨income_effects⁩

Thanks!

**Comment by anjoung** [comment]
I can definitely do this @jackblun!

**Comment by anjoung** [comment]
@jackblun I am running into a bunch of cases where I have two games that are likely two parts of the same game. For example, with Powerball, we might have separate IDs for "Powerball" and "Power Play". Another more extensive example is the game "3D" and "4D" have a bunch of different versions like "3DD", "3D Extra", "3D Digital", etc. Is it best to leave these games separate for now, or should I try to combine them?

**Comment by jackblun** [comment]
To be clear, you only need to combine if they are within-state. For example, if there is a 3DD in NY and a 3D Digital in TX, you don’t need to worry about whether these are the same game or not. If they are in the same state, I wouldn’t combine anything that isn’t clearly the same thing. The cases we want to combine are games where for example in one year in one state they are called “hi5”, then the following year in that same state they are called “highfive”. These should be fairly obvious in most cases. In no case should we be merging games where there are simultaneous sales within state within week - its just the cross-year slight changes in wording / abbreviation that we are after.

**Comment by anjoung** [comment]
I've tried my best to clean-up the game names! I tried to make the changes as transparent as possible, so there is a large chunk of games that I just hand-coded. 

As I was doing this, I realized that some of these game names are actually concatenations of two games. For example, the game "txmillionstx2step" is the combination of "Texas Millions" and "Texas Two Step". What I believe happened is that at some point Texas Millions transitioned into Texas Two Step (the latter being discontinued). This potentially happens several times, and those cases remain improperly coded with regards to both their name and how sales data is being recorded. In three cases, there were notes or formatting in the spreadsheet that made it clear how to deal with this. Those cases were:
- cashwinfalllucky4life ("cashwinfall" and "lucky4life")
- jppokerallorno ("jackpotpoker" and "allornothing")
- lottosouthwin4life ("lottosouth" and "win4life")

Here are all the cases that I identified but could not correct:
- cash4liferolldown
- cash4lifewta
- cash5rdown
- txmillionstx2step
- hotlottorolldown
- hllottoam
- raffleddplay
- rafflehlotto
- rolldownhotlotto
- run4rolldown
- szallstarb
- tictac3llotto
- wflmball
- wincashlotto
- wta2by2

Additionally, I have the following questions for La Fleurs:
- In MT 2016, there is a column for Big Sky Bonus but nothing for Big Sky. However, in MT 2017, we just have a column for Big Sky. Are these both the same game is there a bonus edition? If there is a bonus edition, are we possibly missing games?
- In OH 2008 - 2014, we observe a game called "EZ play"; in OH 2015-2016, we observe the game "EZ P&T". Are these the same game?
- In WI 2017, we observe a game called "Fee". Is this actually a game or something else?
- In IL 2010, we observe a game callled "Mplier SS"; in IL 2009 - 2010, we observe the game "MM Sub". These appear to be related to the Mega Millions game. However, there exists a column for the Megaplier. What do these additional columns denote?

All these questions and notes will be added to the memo.

**Comment by jackblun** [comment]
Thanks @anjoung. To make sure I understand, is the issue that in these cases with two games, the sales for both games are listed in the same column, often without indication of which weeks refer to each game? If so, bar the ones you can easily correct, we should drop these games for each year that this is an issue in, because the switching of games mid-year will just add noise to our data.

La Fleurs have found some (though not much) of the data which we are missing. They sent over three files, which I've added to the La Fleurs folder of Externals in our Dropbox under "additional data 021319". Could you update the cleaning code to read in this data for the appropriate state-years? Thanks again.

**Comment by anjoung** [comment]
@jackblun That's exactly the issue! I can definitely drop those observation and I'll also take a look at the new data!

**Comment by anjoung** [comment]
Added the new La Fleurs data

**Comment by huntallcott** [comment]
@jackblun,
Does your post in #20 mean that you now think that the data are fully cleaned? If so, I’d love it if you could post some diagnostics just to make sure. Here are two proposals:
   - One figure for each state: all games in the state, total sales by quarter, one line for each game, y-axis is a log scale, x-axis is quarter. Make sure that the line stops (instead of interpolating) if there is a missing data point in a given quarter.
   - One figure for each of the multi-state games (MM, PB, ??): total sales by quarter, one line for each state, y-axis is a log scale, x-axis is quarter. Again please make sure that the line stops with missing data.

It seems like we should be able to stare at the resulting ~30 figures and have a very good sense of how clean the data are. 

These figures could stay in pdf form (directly output by stata, you don’t need to pull them through lyx) and should live in a folder called Lotteries/Issues/22.

Does that sound reasonable? And if you have other diagnostic tables and figures re: cleaning, you could put them in that folder as well for all of us to review?
Many thanks.

**Comment by jackblun** [comment]
@anjoung I've uploaded more cleaning stuff, including dropping game-state-years where the data looks no good. The big thing missing is the 94-97 data. How are you getting on with that? We should really wrap up the cleaning of this soon as I'd like to move on to redesigning the experiment, as discussed last week. 

@huntallcott I've produced some diagnostic plots in Issues/22 which look okay to me, but you might spot something I'm missing.

**Comment by anjoung** [comment]
@jackblun The data from 1994 to 1997 has been added and the main cleaning code is edited. The remaining items on that front is identifying game-specific missing data (which I will add to the memo later today).

The only data error that I want to flag is IL1 996w37 - 1996w52. The overall sales variable during that time period does not equal the sum of the columns (though I see we have code that now takes care of that). Additionally, Column F has no label. Other than finding missing data, adding the 94-97 data is done!

**Comment by jackblun** [comment]
Hey @anjoung thanks for sorting out 94-97. That data is now in the analysis and all working.

If you look in Issues/22 here, you'll see plots of all the states' data. This is quite revealing. There are some games which are the same but which you haven't picked up. You did a great job on the others, and these ones are sometimes a bit less obvious. For example, there is a lotto and an fllotto in FL which look the same. These graphs are helpful here since you can see clearly if its just a straight switch in terminology at one point in time. Could you go through all the state graphs in that folder to check for more cases like this, and add to the others in the cleaning file?

Thanks again and let me know if not clear.

**Comment by anjoung** [comment]
Got it! Just took a look and the graphs are pretty clear, so I can definitely start working through them.

**Comment by anjoung** [comment]
@jackblun Where is the code that made those graphs? Would it be okay if I updated those graphs in Issue/22 given some of my updates? Just want to do another round of checking and looking at a slightly more cleaned up set of graphs would be helpful.

**Comment by jackblun** [comment]
@anjoung sure, the code is in a temp folder in Code/

**Comment by anjoung** [comment]
Just finished looking through the graphs and editing the code. To be clear, I only merge together two games if there is no discontinuity between them. Even if I am pretty sure that it is the same game, if game A has data for 2007 and 2011, and Game B has data for 2009, then I do not merge these games together. 

For the ones that I felt were pretty clear and continuous, I added them to the section of cleaning code labeled "Individual Fixes". There were some for which I was not sure. In these cases, the data fits nicely and continuously, but given the assigned names, while it is plausible they are the same game, it is also plausible that they are not. These games are located in the section labeled "Not 100% sure". I'll email state lotteries to see if they can give us some direction on this.

Also, in the Florida Lotto case, we probably need to ask La Fleur's for help. Alongside _fllotto_ the game _fantasy5_ is missing data over an identical time period. On top of this, while the game _lotto_ fits the gaps for both games, so does the game _cashlotto_. I am pretty sure these four games correspond with each other, but not sure in what way. 

Finally, I noticed that there are a few state lotteries that merge nicely with _lottosouth_. This is because in 2001 several state lotteries were merged together to make this lottery. I did not merge these lotteries together in the data, since it seemed like the unmerged lotteries and distinct from the merged lottery.

Updated all the graphs and code, and pushed it. Let me know if you notice anything weird or want me to take another swing!

**Comment by anjoung** [comment]
@jackblun Just updated the data inconsistencies memo using the new data with revised game names and data from 1994 to 1997. Specifically, updated what year-weeks for which we have missing data across the entire state, and missing data at the game-state level. 

One thing that I want to flag is that adding the data from 1994 to 1997 has introduced more "duplicate dates" issues. Previously, we occasionally had duplicate dates on the 52nd week of some years. Now, we have across different weeks between 1994 and 1997, which might make dealing with this issue harder. (for example, the duplicates date issue emerges for AZ 1994w26).

**Comment by jackblun** [comment]
Hi @anjoung , thanks for the update. 

1. On your discussion of the potential game name changes for florida, is it not possible to work out which games are the same by just checking for continuity in the sales levels across the break? To me, it looks like fllotto and lotto line up perfectly, whereas for fantasy5 the years don't match up. Where possible, we want to match games that are the same. Our strategy relies on within-game variation, so we lose a lot of power by separating games that are the same.

2. On the duplicate dates issue, I inspected AZ 1994w26. It looks like one week has been split into two. This is a relatively easy fix - you can just sum sales across the two to get the week total. If you inspect each case, you should be able to rectify issues such as this.

**Comment by anjoung** [comment]
Got it @jackblun. Will try to look more closely for discontinuities to match things up and let you know about any remaining cases. Also, I'll just sum together state-game sales data at the year-week level.

**Comment by jackblun** [comment]
Just very briefly, rather than summing all duplicate weeks you should
inspect at least a few of them (all if not too many) to check whether it is
just the case that they are one week which has been split up. If the cause
is some other problem, such as entirely duplicated rows, clearly we don't
want to sum, but rather drop one of the duplicates. Thanks

**Comment by anjoung** [comment]
@jackblun Took a closer look at all the games that I wasn't 100% sure about, and thing look alright. I didn't use the log_sales_pc variable that you used for your graphs. I just plotted the raw sales variable. There was only one case where things seem to match up, but the noise pattern for one game name is very different than the other (made note in cleaning data; AZ fantasy5 and pick5).  

For the sales data, I identified 4 mislabeled dates and hand-corrected them in the spreadsheets (original raw data left untouched). Then, I just summed sales data at the state-game-week level, which took care of the remaining issues. 

I also updated the graphs that you have in the subfolder Issue/22. I looked them over and didn't see any issues. Let me know if there's anything else on this front! Otherwise, I am going to focus on merging in more state prize structure data with its corresponding sales data.

**Comment by jackblun** [comment]
@anjoung thanks for all this, the graphs are looking much better.

@huntallcott, @bblockwood and @dtaubinsky, before the next round of income effects analysis, now would be a good time for you to take a look at the data diagnostics to see what you think of the La Fleurs data. As before, these are in Issue/22, and split into three subfolders, as described below:
1. sales_gamelevel_bystate contains game level log sales data for all states. This has been useful for inspecting data inconsistencies, as indicated by big jumps in the series. Note the presence of many short-lived games as discussed before. In the analysis, I would propose we restrict to games where we have at least 10 years of data, as its hard to see how we would learn anything at the game level from these short games.
2. multistate contains plots of log sales over time for mm and pb, across states. Here you can clearly see the introduction of cross-selling in 2010, where the powerball states start additionally selling megamillions, and vice versa.
3. sales_sumstats contains several histograms and a table summarizing the game level data. Note here the presence of some games with exceptionally high sales, for example vlt in RI, which has an annual sales per cap of more than $1200. Since these vlt games are in casinos, I think its quite likely that a lot of these players are coming from out of state, hence inflating the per capita levels. Any analysis we run should be run with and without these vlt games. "log_sales_pc_hist_bygametype.pdf" is particularly informative. This shows that the highest sales games are all vlt, followed by instants. I've copied that figure here:
[log_sales_pc_hist_bygametype.pdf](https://github.com/huntallcott/Lotteries/files/2914514/log_sales_pc_hist_bygametype.pdf)

**Comment by huntallcott** [comment]
Thanks much @jackblun @anjoung. I can see that it's been a lot of work.

In response to @jackblun's comments:
1. Minor but log_sales_pc_hist.pdf suggests to me that 5 years might include a lot more games, and I think 5 years is sufficient.
2. Interesting
3. So you're saying that these outliers in sales_pc_hist.pdf are correct and generally RI VLT? 

Some additional thoughts:
A. Given that lottos are gradually dying and instants are gaining, I think we need game type by quarter-of-sample fixed effects in our main regressions, so we identify of differences in lotto purchases in states with growing vs. shrinking income. Note that we may still concerned about unbalanced panels entering those fixed effects (i.e. the residuals will have a jump if a big or small lotto game enters or exits in a given quarter.)
B. In order to implement (A), I think it would be good to break out the "other" category of game.type to include lotto games. Is there any other substantive type of game in the "other" category? If it's easy, I think it would be useful to break out Keno as well. 
C. Still a lot of games that come in for a year or two and then exit. Can you remind me again: what share of this is real vs. data errors? Or put differently, how much do you believe the levels and trends in the *state totals*?
D. Still some seemingly missing data. E.g. play3DE in 2010, instIA, pulltabsIA, pick3IA, and powerballIA in the mid-2000s, HoosierlottoIN around 2010, everthing in KS after 2009, pick3KY around 2010, pick4LA, lottoLA, and instLA in 2010, pick3MD around 2009, cashlottoME in 2003, etc, etc, etc.. These are obviously less than ideal, especially given that a good portion seem to happen between 2007-2012 when we have a lot of income variation. Have you done everything to fully track this down with LaFleur? And can you remind me again: what share of "missing" generally means that the game was not offered for that specific period vs. LaFleur didn't track the data? 
E. /sales_sumstats/log_sales_pc_hist.pdf may be mis-labeled?

Thanks again for all this great work.

**Comment by jackblun** [comment]
@huntallcott in response to the first points:

1) Agreed, whatever we do we should do for a variety of cutoffs of number of quarters required to be included.
3) There are four observations in sales_pc_hist.pdf with sales per cap over $100 per quarter. These are all VLT games. The top two are in RI, the third and fourth are MA and WV. As I was trying to illustrate with log_sales_pc_hist_bygametype.pdf, VLTs tend to have very high sales per cap. [Here](https://www.sisal.com/eng/offering/games/gaming-machines/video-lottery-terminals) is some more info on VLTs. They tend to be located in casinos, and to me seem quite distinct from other lotteries and much closer to other forms of gambling.

And the second:
A) Thats a great suggestion, will implement.
B) Keno is the obvious one. Theres also Pulltabs, which to me seem a lot like instants. With VLT, instants, Keno, Pulltabs, mm and pb pulled out, most of the remainder will be lotto style games.
C) I've searched and it does seem common in reality for games to enter for a year or two then be removed. Do you have a suggestion for a method to assess whether this is the case? There is of course the brute force approach, where we search individually for each game to try to work out when it was sold. This would be possible for the more recent games, but for older games I think the information is not likely to be available. I don't believe the state totals are super accurate.
D) I'll request again from La Fleurs. For individual games within states where they have other data for that year, I wouldn't expect much. The error here is likely in the data submitted from the states themselves rather than La Fleurs. For entirely missing state years, we've had a bit more luck, although I've already pushed them on this. @anjoung can you point me towards the latest version of the document going through the data deficiencies? And does this include cases such as Powerball in CO in 2010? i.e. those where it is clear that the game was running that year, but its just not in the data?
E) Fixed. Thanks.

**Comment by huntallcott** [comment]
@jackblun 
2. Just checking -- that article is from Europe. You're sure that VLT also tends to be in casinos in the US? Absent the EU regulation, it seems like they could also be in convenience stores, airports, etc., at least in theory.

B. Just to be clear -- I think we need to explicitly pull out "lotto" games, not leave them in "other." The reason is that state-run lotto specifically seems to be on a downward trend.
C. I don't have a suggestion other than searching individual cases online. Once you have a sense that this does happen in reality, I don't think you should search more. 
   - Why don't you believe the state totals if you do believe that at least some of these short-lived games are correctly reported as short-lived?
D. OK thanks for trying one more time. 

Thanks again. I do think we're very close to a place where I will believe whatever estimates we get next.

**Comment by jackblun** [comment]
@huntallcott 
2. Sorry for the lazy link. I looked at a few cases but I've had a more thorough look and the location of VLTs varies a lot by state. In RI for example, they are just in the [casinos](https://www.rilot.com/en-us/table-games.html). In WV, they can be in [bars and restaurants too](https://www.americancasinoguide.com/casinos-by-state/west-virgina-casinos.html). New York recently put them in [bars](https://www.crainsnewyork.com/article/20120629/HOSPITALITY_TOURISM/120629866/new-york-bars-add-gambling-terminals). Oregon has them everywhere, even in [convenience stores](https://www.oregonlive.com/politics/2013/11/oregon_lottery_slot_machines_c.html).

B) Agreed, the issue will be that while most keno games are called 'XXkenoYY', there could be some that have non-descriptive names, like 'fantasywinnings'. By picking out a few of these, so far I've found that anything thats not clearly labeled 'vlt', 'instant', 'keno' or 'pulltabs', is a conventional 'lotto' game. However it may be that there are some cases where games are some other funky type of game. An example is [printnplay by mn](https://www.mnlottery.com/games/print-n-play/ppnp-crossword-3).
C) I believe that the majority of these short-lived games are reported correctly. The reason why I'm skeptical of the state totals is that some of the larger, longer-running games are missing years, which will have a bigger effect on the state totals. An example is powerball in CO. I'm not saying the state totals will be way off, but there will definitely be a little bit of noise in them due to games occasionally being omitted from the totals.

**Comment by anjoung** [comment]
@jackblun The memo is in the "Notes" subfolder and is called "Data inconsistencies". Noticed that the specific case you raised is missing from my memo, and am going to figure out why I missed it. It will be completely updated by the end of today/early morning tomorrow.

**Comment by anjoung** [comment]
@jackblun Just looked over the data and my memo should be up-to-date, now. I've been using the dataset "lafleurs_clean" to do all my checking, but saw that there is code in the do-file "dataprep_merge" that suppressed some of this sales data because you observed "weirdness". Since that code only affected the data file "incomeeffects_gamelevel_clean", I did not observe it. From now on, I'll only observe that data when looking for missing data (if that's better).

[Data inconsistencies.docx](https://github.com/huntallcott/Lotteries/files/2920738/Data.inconsistencies.docx)

**Comment by jackblun** [comment]
Hey @anjoung, in the cleaning code, I've gone through and added a variable called 'poorsalesdata_flag'. This is set to 1 if we have found a problem with the data. This is fed both from dataprep_lafleurs and dataprep_merge. The goal of this is to keep clear which data from LaFleurs is actually missing, and which data we have a problem with. They have not being super helpful so we are going to have to be more precise than currently in the data inconsistencies doc.

Approximately 2% of the sales data given to us by la fleurs is flagged as being suspicious or unusable right now. The target is to reduce this as much as possible, and to get them to dig out any missing data they have sitting on hard drives.

Based on this and your data_inconsistencies.doc, could you produce a document with the following sections:

**1. Missing data**

Individual games, or entire state-years which are entirely missing for periods of the data. Please do not include states where there were no lotteries running. Most (perhaps all) of this will be in the data inconsistencies doc you have prepared.

**2. Implausible / unusable data ** 

Any games which we have flagged as being of poor quality. For each of these cases, there should be a description of exactly what the problem is. This may be something like 'sales tripled between december 2010 and january 2011 and then plateaued, making us suspicious that the data is correct'. Or it could be a description of a slight game name change event, and why it is not possible to work out which game changed to which.

Hope this is clear, let me know if not. This should be our final push to sort out the data issues. If its not immediately obvious why some of the data is flagged as poor quality and it is not labelled clearly in the code, let me know, then fill in a comment in the code indicating why that game is flagged as poor data. Thanks for all the hard work.

**Comment by anjoung** [comment]
@jackblun Still have a few more games to observe, but close to done and should 100% have it done on Monday. In the meanwhile, I wanted to post the memo on here. The only thing remaining is commenting on "poor" sales data for each specific game. Everything else is in place, but if you want me to adjust or add anything, please let me know!!
[Data inconsistencies_v2.docx](https://github.com/huntallcott/Lotteries/files/2947639/Data.inconsistencies_v2.docx)

**Comment by anjoung** [comment]
@jackblun Here's the complete memo! I am traveling for grad school visits this week, but if there are any issues just let me know. Will have a lot of lag time while traveling. 

I paid particular focus to Mega Millions and Powerball in this memo. I can do the same for any other games if you'd like. Also, for games we flagged for poor data quality, I highlighted any games that I wasn't completely sure on, or disagreed with. Like I said, let me know if there's anything missing or in need of an edit.
[Data inconsistencies_v2.docx](https://github.com/huntallcott/Lotteries/files/2957675/Data.inconsistencies_v2.docx)

**Comment by jackblun** [comment]
Thanks for this @anjoung . I've reviewed the doc and have a few comments:
1. Raffles are frequently missing for periods within the year. I've done some research and as outlined [here](https://www.lottoexposed.com/lottery-raffles-exposed/), Raffles tend to be short-term special events over holidays. Therefore I think them being missing is probably not a data problem but rather the true state of the world. We should consider this a different category of game, so this has been good to note. The connecticut game "CSD" is also a [raffle style game](https://portal.ct.gov/-/media/DCP/News%20Releases%20from%20the%20Department%20of%20Consumer%20Protection/CTSUPERDRAW). Another example of a game where the data is actually probably correct is "Sports Action", which is a fantasy sports game so likely seasonal also. There is also 'special draw' in Wisconsin which is similar. I've updated the doc to remove these being labeled as inconsistencies. Could you check through for other cases? Its embarrassing for us if we tell La Fleurs they are missing something, but then they come back to us with evidence that a game didn't actually run in that week, which would have been easy to do with the raffle games for example.

2. This doc is very thorough and useful for us internally, but is not something I can send to La Fleurs as there is too much information. I see that we need two docs, this one for us to use and another one which is a request to La Fleurs. Could you restructure and condense this, prioritising the biggest issues. For example, you might just have at the start a list of state-years where we have no data but we know the lottery existed. Then move on to a list of periods where individual games are missing but we are confident the game exists. Then a list of other misc issues, such as columns being merged. In particular, we shouldn't have notes on what we have done in data processing, as this is irrelevant for La Fleurs.

I know this is a lot of work, thanks for all the detail and hope enjoying touring grad schools. I'm waiting to hear from La Fleurs re more general questions on data quality.

**Comment by anjoung** [comment]
Got it! I suspected that I'd have to condense my original document. I'll start working on identifying these "special cases".

**Comment by jackblun** [comment]
Hey @anjoung I'm going to have a call with La Fleurs later this week, is there any chance an updated and condensed doc will be ready say tomorrow so that I can send to them?

**Comment by anjoung** [comment]
Hey @jackblun I can get this to you by the end of tomorrow. What should I do about cases where I perceived a possible issue or had a disagreement with you over coding? Should I just defer to your original judgement and remove my notes?

**Comment by jackblun** [comment]
I had a look at a couple of those cases and I think you can go ahead and over-rule what I did. I’ll check all the cases again later. Just keep a note. Thanks a lot for doing this so quickly.

Jack

**Comment by anjoung** [comment]
@jackblun Finished the memo, but please let me know if there's anything missing or you'd like me to change the formatting. Refer to "Data Inconsistencies_v3" to see the current state of the data. The memo "Data Inconsistencies_v2" is identical to v3, except it retains my notes about spots where you and I had discrepancies. The memo "Memo for La Fleurs" is the memo for La Fleurs.

I noticed one other game that looked seasonal "[scoreboard](http://www.naspl.org/nasplmembers/Oregon)": a sports betting game. Also, there's a game that I suspect is seasonal or "one-off" (WI _supermatch_) but I couldn't find any information about that game. Additionally, I was wondering if I should flag in the data all the games that you had identified as seasonal (e.g., _csd_); would just use the poor data indicator we already have.

[Data inconsistencies_v2.docx](https://github.com/huntallcott/Lotteries/files/2985517/Data.inconsistencies_v2.docx)
[Data inconsistencies_v3.docx](https://github.com/huntallcott/Lotteries/files/2985518/Data.inconsistencies_v3.docx)
[Memo for La Fleurs.docx](https://github.com/huntallcott/Lotteries/files/2985519/Memo.for.La.Fleurs.docx)

**Comment by jackblun** [comment]
Thanks, I’ll take a look at this later. Re the seasonal games, don’t mark them as poor data, as the data is fine. I think we can leave them unflagged.

**Comment by jackblun** [comment]
On that final point, if possible do make a note of the types of names which correspond to seasonal games, in case we forget.

**Comment by jackblun** [comment]
This looks great. Could you explain what the final section, named ‘missing games (unclear cases)’ means? I don’t understand the description.

**Comment by anjoung** [comment]
Those games are game names that I wasn't sure represented a game but had missing data issues. For example, some of those names represent a category of games (e.g., _inst_, _breakopens_, _fastplay_). Other games might be feature of game (e.g., _extra_, _addaplay_). In one specific case, I flagged a game called _megamultipler_ because I am very confident that this game has nothing to do with Mega Millions, but it has such a similar name to the "Megaplier" that I am worried that it was just labeled incorrectly.

The idea was that these "games" have missing data, but since the categories are weird, I wanted to separate them out from games with missing data where it is more clear cut.

**Comment by jackblun** [comment]
Okay, I think I see. Thanks

**Comment by huntallcott** [comment]
@jackblun I think we're going to have @anjoung proceed with the LaFleur data, income effects estimation, etc., so as to keep pressing forward.
Did you end up talking with LaFleur last week? And is there any info you need to transfer to @anjoung so that he can close this out?
Many thanks.

**Comment by jackblun** [comment]
Hi @huntallcott @anjoung , no response from La Fleurs for a week. I sent them the detailed doc on the inconsistencies in the data. When they do get back to us, they tend to be enthusiastic, so I hope that they are digging into the inconsistencies for us. I think that there is still a chance they could find some additional data for us, but its unlikely that the final, cleanest data will be significantly different from what we have now.

**Comment by huntallcott** [comment]
@anjoung @jackblun,
This I think is the main task that we need to transfer over from @jackblun to @anjoung. Per the original [post](https://github.com/huntallcott/Lotteries/issues/22#issue-401396769), the goal was to to clean the LaFleur's data.
Here is my proposal. We close this out this week with the following deliverables:
1. A web link to the code on our repo required to clean the LaFleur data from beginning to a dataset of revenues with key = [state quarter game], and another dataset of gametype with key = [state game]
2. Those datasets finished on dropbox.
3. A google doc linked on this thread that provides a written overview of the data that we have, how the data are cleaned, and any remaining problems with the data. This will eventually be an appendix to the paper.

@anjoung: unless @bblockwood objects, can you spearhead these deliverables over the next several days, asking @jackblun for help only if needed? Would be really fantastic.

**Comment by anjoung** [comment]
@huntallcott My only question is when you say "key = [state game]", is it implied that "key = [state game year-week]" or do you mean a dataset where I collapse all the sales data down to the state-game level?

**Comment by huntallcott** [comment]
For what I mean by "key," see chapter 5 [here](https://web.stanford.edu/~gentzkow/research/CodeAndData.pdf). 
But I think you had a more specific question. I meant the following: the original classification of gametype that you are doing should be for each game in each state, so it doesn't vary by year. So there should be a data input spreadsheet that you have that lists each game x state and how it's classified, right? Eventually, yes, this gets merged into the state x quarter x game final dataset, but originally it has key = [state game].

**Comment by anjoung** [comment]
I think that I understand. By _gametype_, do you mean the kind of lottery ticket? I am not exactly sure what you mean by "_original classification of gametype_". As far as I understand, original raw data does not classify each game. Sorry for the back forth and the confusion on my end.

**Comment by huntallcott** [comment]
Whatever you guys are doing to construct the mm, pb, vlt, inst, pulltab indicator variables, and other game types being discussed in point B [here](https://github.com/huntallcott/Lotteries/issues/22#issuecomment-468504193), above on this thread.

**Comment by anjoung** [comment]
Got it! Thank you

**Comment by anjoung** [comment]
Update! To classify by game type each game-state pair, I've been moving one-by-one looking up each game. Some games I can just easily classify because the name makes it clear (e.g., _clubkeno_=keno, _lottoamerica_=lotto). However, a lot of names don't give any clues. In previous comments, it was suggested that Jack or I just quickly scan a few and see if most of these are just lottos. It looks like 90%-95% of them are lottos, but there is a small portion that are not. My plan is to basically look-up each state-game pair with a non-obvious name and classify it manually. 

Before I do there are a few questions that I have.

1. **How do I deal with rapid fire lotteries?** There are some lotteries that run a draw every 4-minutes. They are identical in structure to a traditional lottery, just faster draws. My impression is that they are marketed as this keno/lotto hybrid, where the structure is a lottery but the pace is a keno. Wondering if these should be categorized as more keno, or just dumped in the "other" category.
2. **What is a keno game?** My impression is that a keno game asks a participant to choose (a) how many "slots" they'd like to potentially match on, (b) select a number for each slot, and (c) earn prizes based on number of matches and number of slots originally chosen. However, I wanted to make sure, since I am running into a few and don't want to mis-classify. Also, I've run into horse racing keno-ish games, where the order of your matches can play a factor. Here's an [example](https://dclottery.com/img/games/betslips/r2r-b.gif). Are these horse race games keno, or a totally different category?
3. **How do I deal with add-ons?** A handful of game names are just add-ons like _bonus_, _multiplier_, or _kicker_. Are add-ons classified identically to the parent game, or do are they in a separate "add-on" category? 

Any game names that I cannot classify, I am going to add to my data problems memo.

**Comment by anjoung** [comment]
Sorry! One more question. My understanding from previous posts was that we should only keep games with at least 5-years of data, but I wanted to double-check that I wasn't misunderstanding.

**Comment by huntallcott** [comment]
1. I'd say these are "other"
2. I'd say to use your best judgment.
3. I'd put add-ons with the parent game.
4. RE: your subsequent post, feel free to leave gametype missing for games with less than 5 years of data, and yes please drop those in the income effects estimation.
5. In terms of any specific games where the classification is questionable or difficult, I would actually indicate that in the spreadsheet/csv file you're developing. (Presumably there is a spreadsheet where you have the game name from LaFleur's and the game type you're assigning it. I would consider adding a secondary column called gametype_secondary to give the second type for these edge cases in 1 and 2, and/or an "Uncertain" column that takes value 1 if you are uncertain. Then add a "Notes" column for your comments, and that column should also include any web links you found about the game that you used to classify the game. Does that make sense? 

Thanks again.

**Comment by anjoung** [comment]
This all makes sense and is very helpful! Will get to work on this.

**Comment by anjoung** [comment]
Here's the spreadsheet where I classify all game-state pairs with at least five years of data:
[Game classification.xlsx](https://github.com/huntallcott/Lotteries/files/3048863/Game.classification.xlsx). This file is located in the "Notes" subfolder in Dropbox.

I've marked everything for which I have some degree of uncertainty. For the following broad categories of games, I just want to make sure that I didn't do anything you disagree with:
1. **3d and 4d games**: I am very sure that all these games are "3-digit" or "4-digit" games, which are just variants of the "pick 3" and "pick 4" games, or "match 3" or "match 4". Either way these games would be lotteries. However, I wouldn't be able to say anything about whether or not the combos and prize structure is "complex".
2. **Lotteries with complex combos**: Most lotteries have some variant of "match these number in order (or any order) and win a prize". However, there is sub-set of lotteries with a more complex prize structure. I classified these as lotteries, but add a sub-classification of "complex". [Example](https://www.delottery.com/Drawing-Games/Play-4/Prizes-And-Odds).
3. **Poker themed games**: There are some lottery games that have a poker theme. For these games, you pick cards (suite and number), and winning combinations are poker hands. These are similar to the lottery games with a complex combination. I classify them as lottos, but subclassify them as "poker games". [Example](https://www.kylottery.com/apps/draw_games/5cardcash/index.html#tab-drawGamesTabs_1).
4. **Horse race games**: There are some ticket games where you pick "winning horses" and "winning times". They are draw games, but I felt that these games were sufficiently complex and different that they do not qualify as lotteries. They are classified as "Horse Race" (which will be collapsed under "other"). [Example](https://dclottery.com/games/race2riches.aspx).


For these individual games, I am unsure of how to classify them, or just want to confirm that my classification isn't weird:

1. **bonusWV**: I am pretty sure that this is an add-on for a keno. This is because I found an example of a game called "Bonus Keno"; also, in the raw data, the sales data for this game is adjacent to a Keno. However, the name "bonus" is fairly ambiguous so I can't say that I am more than 90% sure.
2. **Lucky Lines**: Categorized this game as "Tic Tac Toe" (i.e., "other"). [Example](https://www.oregonlottery.org/games/draw-games/lucky-lines).
3. **My Day**: Categorized this game as a lottery, but arguable should be categorized as other. The theme is very unique and affects the prize structure, but overall the prize structure is very similar to a complex lottery. [Example](https://m.nelottery.com/homeapp/lotto/33/gamedetail).
4. **Pick 5 AZ**: This remains unclassified. I have no info about Pick 5, but think that I could just assume that it is identical to Fantasy 5. Arizona changed Fantasy 5 to "Pick 5" and then later changed it back to Fantasy 5. I don't have enough information to show whether the prize structure changed between name changes. I can merge sales data between Fantasy 5 and Pick 5 without discontinuities, so on the surface things look safe.
5. **Quick Cash GA**: This remains unclassified. No information found. Other state with quick cash game is labeled as lotto.
6. **Win 4 Life NY**: This remains unclassified. Most likely a lottery since all other games with this name are. However, there is a currently available scratch-off with this name. I don't think that this is a scratch-off because of the available date range of our sales data for this game.

**Comment by huntallcott** [comment]
@anjoung this looks good. 
A. Since this will be used for the data analysis, please go ahead and move it to the Externals/Data folder? B. Then please develop code to input this spreadsheet into Stata and merge it into the LaFleur's data, keeping the `type` variable, and then using type x quarter-of-sample as a control in the income effects regressions?
C. So it seems like the remaining thing from my [beginning-of-week post](https://github.com/huntallcott/Lotteries/issues/22#issuecomment-478369701) is to finish off the state-by-game-by-quarter cleaned data from LaFleur's? 
D. Sorry I'm telling you what to do all the time even though Ben is the boss. @bblockwood is this OK with you?

**Comment by bblockwood** [comment]
Thanks @anjoung! @huntallcott, we are aligned on all of this. By all means please do weigh in like this so we keep the ball moving quickly on this project -- I will too, and will raise alarm if you write anything I disagree with ;-)

**Comment by huntallcott** [comment]
Hi @anjoung, just checking in, how is all this going?
Would be great to move onto the analysis, that's the more fun part, once the data cleaning is wrapped up. 
Thanks for your work.

**Comment by bblockwood** [comment]
Sorry this was a communication gap on my end — @anjoung was offline Monday through today for grad school visits, with my blessing.

**Comment by anjoung** [comment]
Wrapped up compiling (1) a dataset with gametype key = [game state], and (2) a dataset with revenues key = [game state quarter-year]. 

For (1), here's the data: [Game Classifications.xlsx](https://github.com/huntallcott/Lotteries/files/3070333/Game.Classifications.xlsx). This is located in the "Externals/Data" subfolder. 

For (2), you need to first run "SetGlobals", but afterwards if you just run "dataprep_incomeeffects.do" (located in the "Code/Build/income_effects" subfolder in GitHub), you will have the cleaned dataset. The final cleaned dataset is called "incomeeffects_gamelevel_clean" which is in the "Externals/Intermediate" subfolder. 

The only remaining item is to wrap up the data memo.

**Comment by huntallcott** [comment]
Great, thank you @anjoung!
Excited to read the data memo.

**Comment by anjoung** [comment]
Here is the data memo: 
[Data inconsistencies (detailed).docx](https://github.com/huntallcott/Lotteries/files/3087065/Data.inconsistencies.detailed.docx)

I also made a version with far less detail, and focuses on data problems: [Data inconsistencies (simplified).docx](https://github.com/huntallcott/Lotteries/files/3087069/Data.inconsistencies.simplified.docx)

Please let me know if there are any changes that you'd like me to make!

**Comment by huntallcott** [comment]
Andrew:
Great, thank you. I didn't do a full check but these are some quick reactions:
1.	Great work on the two memos. They are similar enough in length (22 vs. 19 pages) that I don’t see what we are getting out of the simplified version. Feel free to delete the simplified one unless I'm missing something? 
2.	Are the comments in dataprep_lafleurs correct? E.g. line 14 says that “Note 1994 data too incomplete to use” yet we have 1994 data in the dataset. I’m confused by this.
3.	Another way to do the poorsalesdata_flag would be to add columns to Game Classifications or to have a separate csv file that lists [state game]  in rows. Then add columns to indicate the start and end of years with poorsalesdata_flag ==1, then has another column with the verbal explanation of why you think there’s an error. Then you import that into stata and merge to the lafleur’s data. The benefit of that is that your list AND your justification are all in the same place, whereas now you have manual stata code in one place and explanations in a separate word doc. At this point I don’t recommend changing, but if something arises like this in the future, I would do that.
4.	In dataprep_merge.do, is it OK to delete the code starting on line 131 labeled “OLD”?

@bblockwood @dtaubinsky let us know if other thoughts, else I think we are set to close this after @anjoung addresses the above.

**Comment by anjoung** [comment]
1. I mainly removed extra details that might distract a reader when scanning over the document, so nothing major. Jack had me create these simplified documents mainly for La Fleurs, so that we send them as condensed a memo as possible. Will remove the simplified document. Can revive it if it seems necessary.
2. Comments are incorrect and outdated. I removed that comment and a few others. 
3. That is definitely a better idea. Will definitely implement that in the future (this project and else where). 
4. Just deleted it!

**Comment by bblockwood** [comment]
Thanks @anjoung, and @huntallcott for input on this. I'm closing this issue; we can reopen as any new LaFleur cleaning needs arise.

**Comment by bblockwood** [comment]
One update — I came across [this Vox article](https://www.vox.com/identities/2016/1/13/10763268/lottery-poor-prey) which plots sales by game type in CT over time:

<img width="737" alt="image" src="https://user-images.githubusercontent.com/956485/56377846-58f8ab00-61d9-11e9-8373-728db3304b5c.png">

For comparison, this is what ours looks like for CT using LeFleur's data:

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/956485/56378730-e0dfb480-61db-11e9-82bc-c19de2ca6dd4.png">

The "daily" type just represents lottos with the word "daily" in their title, so misclassification might explain the difference from 2009 onward. Overall, the picture looks quite similar. (Code [here](https://github.com/huntallcott/Lotteries/blob/master/Issues/22/benchmark%20against%20CT/CTbenchmark.do).) 

Finally, here's what that graph looks like for the whole country. The shift from lottery tickets to scratchoffs over the past 20 years is quite pronounced. 

<img width="1067" alt="image" src="https://user-images.githubusercontent.com/956485/56378838-3a47e380-61dc-11e9-89ff-9bc6fbc73010.png">

**Comment by bblockwood** [comment]
Update: the data behind the Vox article is [here](https://data.ct.gov/Government/Lottery-Sales-By-Game/n9q9-yvd6).
